chains/substrate/config.go

```diff
@@ -1,39 +1,54 @@
 package substrate
 
 import (
-	"strconv"
+	"fmt"
+	"math/big"
 
 	"github.com/ChainSafe/chainbridge-core/chains"
+	"github.com/spf13/viper"
 )
 
 type SubstrateConfig struct {
 	GeneralChainConfig chains.GeneralChainConfig
-	StartBlock         uint64
+	StartBlock         *big.Int
 	UseExtendedCall    bool
 }
 
-func ParseConfig(rawConfig *chains.RawChainConfig) *SubstrateConfig {
-	config := &SubstrateConfig{
-		GeneralChainConfig: rawConfig.GeneralChainConfig,
-		StartBlock:         0,
-		UseExtendedCall:    false,
+type RawSubstrateConfig struct {
+	GeneralChainConfig chains.GeneralChainConfig
+	StartBlock         int64 `mapstructure:"startBlock"`
+	UseExtendedCall    bool  `mapstructure:"useExtendedCall"`
+}
+
+func GetConfig(path string, name string) (*SubstrateConfig, error) {
+	var config RawSubstrateConfig
+
+	viper.AddConfigPath(path)
+	viper.SetConfigName(name)
+	viper.SetConfigType("json")
+
+	if err := viper.ReadInConfig(); err != nil {
+		return nil, fmt.Errorf("failed to read in the config file, error: %w", err)
 	}
 
-	if blk, ok := rawConfig.Opts["startBlock"]; ok {
-		res, err := strconv.ParseUint(blk, 10, 32)
-		if err != nil {
-			panic(err)
-		}
-		config.StartBlock = res
+	if err := viper.Unmarshal(&config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config into struct, error: %w", err)
 	}
 
-	if b, ok := rawConfig.Opts["useExtendedCall"]; ok {
-		res, err := strconv.ParseBool(b)
-		if err != nil {
-			panic(err)
-		}
-		config.UseExtendedCall = res
+	if err := config.GeneralChainConfig.Validate(); err != nil {
+		return nil, err
 	}
 
+	parsedCfg := parseConfig(&config)
+
+	return parsedCfg, nil
```

Comment:
The pointer here has been left off. See above comment; different declaration/initialization convention of using struct literal helps with not forgetting the pointer

`&parsedCfg`